core/src/main/java/com/gentics/mesh/core/endpoint/node/S3BinaryMetadataExtractionHandlerImpl.java

```diff
@@ -0,0 +1,332 @@
+
+	private final Vertx vertx;
+
+	@Inject
+	public S3BinaryMetadataExtractionHandlerImpl(Database db, S3BinaryStorage s3BinaryStorage, HandlerUtilities utils,
+												 Vertx rxVertx, MeshOptions options, S3Binaries s3binaries, S3BinaryProcessorRegistryImpl s3binaryProcessorRegistry) {
+		this.db = db;
+		this.vertx = rxVertx;
+		this.s3BinaryStorage = s3BinaryStorage;
+		this.utils = utils;
+		this.options = options;
+		this.s3binaries = s3binaries;
+		this.s3binaryProcessorRegistry = s3binaryProcessorRegistry;
+	}
+
+	/**
+	 * Handle a request to create a new field.
+	 *
+	 * @param ac
+	 * @param nodeUuid   UUID of the node which should be updated
+	 * @param fieldName  Name of the field which should be created
+	 * @param attributes Additional form data attributes
+	 */
+	public void handleMetadataExtraction(InternalActionContext ac, String nodeUuid, String fieldName, MultiMap attributes) {
+		validateParameter(nodeUuid, "uuid");
+		validateParameter(fieldName, "fieldName");
+
+		S3BinaryMetadataRequest request = JsonUtil.readValue(ac.getBodyAsString(), S3BinaryMetadataRequest.class);
+		String languageTag = request.getLanguage();
+		if (isEmpty(languageTag)) {
+			throw error(BAD_REQUEST, "upload_error_no_language");
+		}
+
+		String nodeVersion = request.getVersion();
+		if (isEmpty(nodeVersion)) {
+			throw error(BAD_REQUEST, "upload_error_no_version");
+		}
+
+		S3UploadContext ctx = new S3UploadContext();
+
+
+		String bucket = options.getS3Options().getBucket();
+		String objectKey = nodeUuid + "/" + fieldName;
+		s3BinaryStorage
+				.exists(bucket, objectKey)
+				//read from aws and return buffer with data
+				.flatMap(res -> {
+					if (res) {
+						return s3BinaryStorage.read(bucket, objectKey).singleOrError();
+					} else return null;
+				}).flatMap(fileBuffer -> {
+			if (nonNull(fileBuffer) || fileBuffer.getBytes().length > 0) {
+				return db.singleTx(tx -> s3binaries
+						.findByS3ObjectKey(nodeUuid + "/" + fieldName)
+						.runInExistingTx(tx)
+						.getFileName())
+						.flatMap(fileName -> {
+							String mimeTypeForFilename = MimeMapping.getMimeTypeForFilename(fileName);
+							File tmpFile = new File(System.getProperty("java.io.tmpdir"), fileName);
+							vertx.fileSystem().writeFileBlocking(tmpFile.getAbsolutePath(), fileBuffer);
+							byte[] fileData = fileBuffer.getBytes();
+							FileUpload fileUpload = new FileUpload() {
+
+								@Override
+								public String uploadedFileName() {
+									return tmpFile.getAbsolutePath();
+								}
+
+								@Override
+								public long size() {
+									return fileData.length;
+								}
+
+								@Override
+								public String name() {
+									return fileName;
+								}
+
+								@Override
+								public String fileName() {
+									return fileName;
+								}
+
+								@Override
+								public String contentType() {
+									return mimeTypeForFilename;
+								}
+
+								@Override
+								public String contentTransferEncoding() {
```

Comment:
Are these two nulls below correctly processed? 
If null is not expected there as a special case, I would place some existing (if possible) or default values there. Charset is usually UTF-8. 
If it is not possible to get these values at the moment, I will test this part explicitly during the final merge.

Reply:
I fixed it for the charset, but I cannot find any default value for the contentTransferEncoding. In any case I am not using these values at all.